corehq/ex-submodules/couchforms/models.py

```diff
@@ -96,9 +96,11 @@ class XFormOperation(DocumentSchema):
     operation = StringProperty()  # e.g. "archived", "unarchived"
 
 
-class XFormInstance(SafeSaveDocument, UnicodeMixIn, ComputedDocumentMixin,
-                    CouchDocLockableMixIn, AbstractXFormInstance):
+class XFormInstance(DeferredBlobMixin, SafeSaveDocument, UnicodeMixIn,
```

Comment:
`DeferredBlobMixin` seemed like a cool idea because it allows attachments added to a form to be persisted in the blob db when the form is `save()`d. However, it also seems to be the source of [this test failure](https://travis-ci.org/dimagi/commcare-hq/jobs/140928493).

Here's my current hypothesis on the issue:
`XFormInstance`s are sometimes [saved with `bulk_save`](https://github.com/dimagi/commcare-hq/blob/3f49c8a5b711d5434dae72f1e26628797e5a76a4/corehq/form_processor/backends/couch/processor.py#L66), which means the [`save` method](https://github.com/dimagi/commcare-hq/blob/3f49c8a5b711d5434dae72f1e26628797e5a76a4/corehq/blobs/mixin.py#L412) is not called, and forms' attachments are not saved. The reason that this previously worked is that the attachments were encoded in the doc dict that was uploaded to couch either with `xforminstance.save()` or `XFormInstance.get_db().bulk_save(docs)`. Note: I found this issue while analyzing the test failure; attachments not being saved may not be the direct cause of the test failure.

@dannyroberts / @snopoke: my current plan is to build some `atomic_blobs`-like thing that works with `bulk_save` and then either override `bulk_save` or track down all the uses of `bulk_save`-like things that are currently saving `DeferredBlobMixin` objects and wrap them with attachment saving logic. The override seems easy enough but expands the code that needs to know about the blob db. The second seems hard and possibly error prone (easy to miss one). Do you have any other ideas of how I could address this issue?

Aside: the use of `bulk_save` in our codebase seems to imply an argument against putting any custom logic in the `Document.save` method because it means it is possible to save an object without executing its custom save logic.